src/main/java/org/mockbukkit/mockbukkit/block/data/AgeableDataMock.java

```diff
@@ -0,0 +1,41 @@
+package org.mockbukkit.mockbukkit.block.data;
+
+import com.google.common.base.Preconditions;
+import org.bukkit.Material;
+import org.bukkit.block.data.Ageable;
+import org.jetbrains.annotations.NotNull;
+import org.mockbukkit.mockbukkit.exception.UnimplementedOperationException;
+
+public class AgeableDataMock extends BlockDataMock implements Ageable
+{
+
+	/**
+	 * Constructs a new {@link AgeableDataMock} for the provided {@link Material}.
+	 *
+	 * @param material The material this data is for.
+	 */
+	public AgeableDataMock(@NotNull Material material)
+	{
+		super(material);
+	}
+
+	@Override
+	public int getAge()
+	{
+		return this.get(BlockDataKey.AGE_KEY);
+	}
+
+	@Override
+	public void setAge(int age)
+	{
+		Preconditions.checkArgument(age >= 0, "Age cannot be negative.");
+		this.set(BlockDataKey.AGE_KEY, age);
+	}
+
+	@Override
+	public int getMaximumAge()
+	{
+		throw new UnimplementedOperationException("Get maximum age is not implemented!");
+	}
```

Comment:
I did not implement this, because we have so many configurations:
![imagem](https://github.com/user-attachments/assets/cd645b64-81a8-4d29-804d-8ec14ca185ec)
So I left it without an upper limit